### PR TITLE
ExecutorServiceSum, ForkJoinSum, MyCallable, Util classes and Tests i…

### PR DESCRIPTION
**Pull request by semyonich**
…mplemented

### CODE REVIEW

**Review by MykolaFatkullin — APPROVED** [review]

File: src/main/java/com/multithread/ExecutorServiceSum.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.concurrent.Future;
+
+public class ExecutorServiceSum {
+    private int numberOfThreads;
+    private List<Integer> intList;
+
+    public Long calculateSum() {
+        int partitionListLength = intList.size() / numberOfThreads;
+        ExecutorService executor = Executors.newFixedThreadPool(numberOfThreads);
+        List<MyCallable> callableTasks = new ArrayList<>();
```
Comment (RomanDubovskyi) [inline src/main/java/com/multithread/ExecutorServiceSum.java]:
```suggestion
        List<Callable> callableTasks = new ArrayList<>();
```
use abstraction where possible

File: src/main/java/com/multithread/ForkJoinSum.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Override
+    public Long compute() {
+        if (intList.size() > THRESHOLD) {
+            return ForkJoinTask.invokeAll(createSubtasks())
+                    .stream()
+                    .mapToLong(ForkJoinTask::join)
+                    .reduce(0, Long::sum);
+        } else {
+            return processing(intList);
```
Comment (RomanDubovskyi) [inline src/main/java/com/multithread/ForkJoinSum.java]:
```suggestion
        } 
            return processing(intList);
```
we don't need else after return

File: src/main/java/com/multithread/ForkJoinSum.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    private Collection<ForkJoinSum> createSubtasks() {
+        List<ForkJoinSum> dividedTasks = new ArrayList<>();
+        dividedTasks.add(new ForkJoinSum(intList.subList(0, intList.size() / 2)));
+        dividedTasks.add(new ForkJoinSum(intList.subList(intList.size() / 2, intList.size())));
+        return dividedTasks;
+    }
+
+    private Long processing(List<Integer> list) {
```
Comment (RomanDubovskyi) [inline src/main/java/com/multithread/ForkJoinSum.java]:
```suggestion
    private Long process(List<Integer> list) {
```

File: src/main/java/com/multithread/MyCallable.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package com.multithread;
+
+import java.util.List;
+import java.util.concurrent.Callable;
+
+public class MyCallable implements Callable<Long> {
```
Comment (RomanDubovskyi) [inline src/main/java/com/multithread/MyCallable.java]:
```suggestion
public class ListSumCulculator implements Callable<Long> {
```

File: src/main/java/com/multithread/Util.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package com.multithread;
+
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class Util {
```
Comment (RomanDubovskyi) [inline src/main/java/com/multithread/Util.java]:
```suggestion
public class ListProducer {
```

File: src/test/java/ExecutorServiceSumTest.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+public class ExecutorServiceSumTest {
+    private static ExecutorServiceSum executorServiceSum;
+    private static final int THREADS_NUMBER_1 = 7;
+    private static final int THREADS_NUMBER_2 = 4;
+    private static final int NUMBER_OF_VALUES = 1000_000;
+    private static final int MAX_LIST_VALUE = 1;
+
+    @BeforeAll
+    public static void beforeAll() {
```
Comment (RomanDubovskyi) [inline src/test/java/ExecutorServiceSumTest.java:16]: add more than one test case, this of different scenarios

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/com/multithread/ExecutorServiceSum.java (hunk 1 above)
Comment (semyonich) [reply src/main/java/com/multithread/ExecutorServiceSum.java]: done

File: src/main/java/com/multithread/ForkJoinSum.java (hunk 2 above)
Comment (semyonich) [reply src/main/java/com/multithread/ForkJoinSum.java]: done

File: src/main/java/com/multithread/ForkJoinSum.java (hunk 3 above)
Comment (semyonich) [reply src/main/java/com/multithread/ForkJoinSum.java]: done

File: src/main/java/com/multithread/MyCallable.java (hunk 4 above)
Comment (semyonich) [reply src/main/java/com/multithread/MyCallable.java]: done

File: src/main/java/com/multithread/Util.java (hunk 5 above)
Comment (semyonich) [reply src/main/java/com/multithread/Util.java]: done

File: src/test/java/ExecutorServiceSumTest.java (hunk 6 above)
Comment (semyonich) [reply src/test/java/ExecutorServiceSumTest.java:16]: done

**Review by RomanDubovskyi — APPROVED** [review]
good job

**Pull request merged**